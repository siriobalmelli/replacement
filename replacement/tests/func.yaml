---
# function execution
# tests/func.yaml
replacement:
  # run a function returning a dictionary, and merge that into 'meta'
  - meta: func
    args:
      existing: {'original': 'thesis'}
    # NOTE that this is sensitive to PYTHONPATH
    input: |
      ret_a_dict ./demo.py
  - text: text
    prep: format
    input: |
      original {original}
      secret {secret}
  # run a function returning a dictionary and export return as JSON
  - text: func
    options:
      - json  # emit JSON rather than YAML (the default)
    args:
      existing: {'original': 'thesis'}
    input: |
      ret_a_dict ./demo.py
  # run a function returning a list of strings; treat each one as a line of text
  - text: func
    args: {}
    # notice non-portable BRAINDEAD python dot.notation
    # ... will break if 'replacement' (the script) is MOVED vis-a-vis demo.py
    # (of OF COURSE the CWD is blithely ignored ... why would ANYONE use it LOL)
    # TLDR: use the '[symbol] [path]' notation in the preceding example
    input: |
      tests.demo.ret_a_stream
  # same, but with a list of text lines (and proper importing)
  - text: func
    args:
      an_arg: ["question"]
    input: |
      ret_a_list ./demo.py
...
