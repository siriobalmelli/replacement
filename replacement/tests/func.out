original thesis
secret 42
{"secret": 42, "original": "thesis"}
1. hello
2. world
42
meaning
question
